core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -217,17 +211,44 @@ static Context deleteContext(Map<String, String> config) {
         Context dataContext = dataContext(config);
 
         String codecAsString = config.get(DELETE_AVRO_COMPRESSION);
-        CodecFactory codec = codecAsString != null ? toCodec(codecAsString) : dataContext.codec();
+        String compressionLevel = config.getOrDefault(DELETE_AVRO_COMPRESSION_LEVEL, AVRO_COMPRESSION_LEVEL_DEFAULT);
+        CodecFactory codec = codecAsString != null ? toCodec(codecAsString, compressionLevel) : dataContext.codec();
 
         return new Context(codec);
       }
 
-      private static CodecFactory toCodec(String codecAsString) {
+      private static CodecFactory toCodec(String codecAsString, String compressionLevel) {
+        CodecFactory codecFactory = null;
         try {
-          return CodecName.valueOf(codecAsString.toUpperCase(Locale.ENGLISH)).get();
+          switch (Codec.valueOf(codecAsString.toUpperCase(Locale.ENGLISH))) {
+            case UNCOMPRESSED:
+              codecFactory = CodecFactory.nullCodec();
+              break;
+            case SNAPPY:
+              codecFactory = CodecFactory.snappyCodec();
+              break;
+            case ZSTD:
+              codecFactory = CodecFactory.zstandardCodec(getCompressionLevel(compressionLevel,
+                  ZSTD_COMPRESSION_LEVEL_DEFAULT));
```

Comment:
We try to make line breaks at the most useful place for readability. This is a bit odd because it's nested 2 levels. Instead, I think it would be better to add the line break so that the `compressionLevel` call is on a single line.

Reply:
Agree it's more readable to have compressionLevelAsInt call on a single line.